### PR TITLE
sysvars: add `tidb_shard_row_id_bits` and `tidb_pre_split_regions`

### PR DESCRIPTION
**Pull request by jackysp**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/18649
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Frank945946 — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -3159,6 +3159,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
 
+### `tidb_pre_split_regions` <span class="version-mark">New in v8.4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
+- Type: Integer
+- Default value: `0`
+- Range: `[0, 15]`
+- This variable is used to set the default row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically set it when executing CREATE TABLE statements, allowing use of pre_split_regions on such tables (for example NONCLUSTERED tables). See [pre_split_regions](/sql-statements/sql-statement-split-region.md) for details. This variable is typically used in conjunction with `tidb_shard_row_id_bits` to shard new tables and split the regions of new tables.
```
Comment (hfxsd) [inline system-variables.md:3170]:
```suggestion
- This variable is used to set the default row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically set it when executing `CREATE TABLE` statements, allowing use of `PRE_SPLIT_REGIONS` on such tables (for example NONCLUSTERED tables). For more information, see [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions). This variable is typically used in conjunction with [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-new-in-v840) to shard new tables and split the Regions of new tables.
```

File: system-variables.md [hunk 2]
```diff
@@ -3159,6 +3159,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - Range: `[1, 100000]`
 - This variable controls the maximum number of execution plans that can be cached by [Non-prepared plan cache](/sql-non-prepared-plan-cache.md).
 
+### `tidb_pre_split_regions` <span class="version-mark">New in v8.4.0</span>
```
Comment (hfxsd) [inline system-variables.md:3162]:
```suggestion
### tidb_pre_split_regions <span class="version-mark">New in v8.4.0</span>
```

File: system-variables.md [hunk 3]
```diff
@@ -5271,6 +5281,16 @@ SHOW WARNINGS;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">New in v8.4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
+- Type: Integer
+- Default value: `0`
+- Range: `[0, 15]`
+- This variable is used to set the shard bits for new table's default row ID. When this variable has a non-zero value, TiDB will automatically set this property when executing CREATE TABLE statements on tables that allow using `shard_row_id_bits` (such as NONCLUSTERED tables). See [SHARD_ROW_ID_BITS](/shard-row-id-bits.md) for details.
```
Comment (hfxsd) [inline system-variables.md:5292]:
```suggestion
- This variable is used to set the shard bits for the default row ID of a new table. When this variable has a non-zero value, TiDB will automatically set this property when executing `CREATE TABLE` statements on tables that allow using `SHARD_ROW_ID_BITS` (such as NONCLUSTERED tables). For more information, see [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).
```

**Review by hfxsd — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -3159,6 +3159,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
 
+### tidb_pre_split_regions <span class="version-mark">New in v8.4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
+- Type: Integer
+- Default value: `0`
+- Range: `[0, 15]`
+- This variable is used to set the default row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically set it when executing `CREATE TABLE` statements, allowing use of `PRE_SPLIT_REGIONS` on such tables (for example NONCLUSTERED tables). For more information, see [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions). This variable is typically used in conjunction with [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-new-in-v840) to shard new tables and split the Regions of new tables.
```
Comment (lilin90) [inline system-variables.md:3170]:
```suggestion
- This variable is used to set the default number of row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically apply this attribute to tables that allow the use of `PRE_SPLIT_REGIONS` (for example, `NONCLUSTERED` tables) when executing `CREATE TABLE` statements. For more information, see [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions). This variable is typically used in conjunction with [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-new-in-v840) to shard new tables and pre-split the Regions of new tables.
```

File: system-variables.md [hunk 5]
```diff
@@ -5271,6 +5281,16 @@ SHOW WARNINGS;
...
 
+### tidb_shard_row_id_bits <span class="version-mark">New in v8.4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
+- Type: Integer
+- Default value: `0`
+- Range: `[0, 15]`
+- This variable is used to set the shard bits for the default row ID of a new table. When this variable has a non-zero value, TiDB will automatically set this property when executing `CREATE TABLE` statements on tables that allow using `SHARD_ROW_ID_BITS` (such as NONCLUSTERED tables). For more information, see [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).
```
Comment (lilin90) [inline system-variables.md:5292]:
```suggestion
- This variable is used to set the default number of row ID shard bits for newly created tables. When this variable is set to a non-zero value, TiDB will automatically apply this attribute to tables that allow the use of `SHARD_ROW_ID_BITS` (for example, `NONCLUSTERED` tables) when executing `CREATE TABLE` statements. For more information, see [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**